cf_cli_java_plugin.go

```diff
@@ -168,10 +182,24 @@ func (c *JavaPlugin) execute(commandExecutor cmd.CommandExecutor, uuidGenerator
 	}
 
 	var remoteCommandTokens = []string{JavaDetectionCommand}
-
+	heapdumpFileName := ""
+	localFileFullPath := ""
 	switch command {
 	case heapDumpCommand:
-		heapdumpFileName := "/tmp/heapdump-" + uuidGenerator.Generate() + ".hprof"
+
+		supported, err := util.CheckRequiredTools(applicationName)
+		if err != nil || !supported {
+			return "required tools checking failed", err
+		}
+
+		fspath, err := util.GetAvailablePath(applicationName, remoteDir)
+		if err != nil {
+			return "", err
+		}
+		dumpFile := applicationName + "-heapdump-" + uuidGenerator.Generate() + ".hprof"
+		heapdumpFileName = fspath + "/" + dumpFile
+		localFileFullPath = localDir + "/" + dumpFile
```

Comment:
As far as I can see we need `localFileFullPath` only in line 260 and only if the parameter `local-dir` was set. Maybe it's worth moving this variable closer to where it's being used (i.e. just before line 260)